### ISSUE TITLE
Error: Invalid Refresh Token: Refresh Token Not Found

### ISSUE THREAD

**Issue by pradeepb28** [supabase-community/supabase-swift#106]
How to handle the following error "Invalid Refresh Token: Refresh Token Not Found" . I want to handle new token each time when the existing token is expired.

**Comment by pradeepb28** [comment]
I've tried following code and it throws error

```
func refreshNewToken() {
        Task {
            do {
                
                let currentAccessToken = try await SupabaseService.shared.client.auth.session.accessToken
                try await client.auth.refreshSession(refreshToken: currentAccessToken)
            } catch {
                print("Unable to refresh session:\(error.localizedDescription)")
            }
        }
    }
```

**Comment by grsouza** [comment]
Hi @pradeepb28 you shouldn't need to refresh the token manually, supabase client internally checks for the token and tries to refresh it in every call if it has expired or is about to expire.

I'll investigate if there's any bug with this implementation.

**Comment by pradeepb28** [comment]
Thanks for your reply. It seems like the issue is in the SDK then.

**Comment by pradeepb28** [comment]
@grsouza is there any ETA to fix the issue? The whole point of SDK is useless if sessions doesn't work as intended because the whole sign-in is not valid at all.

**Comment by adurov9** [comment]
Hello guys, I was having the same issue and this really helps me to find out where the issue is. I hope it helps you too.
https://discord.com/channels/839993398554656828/1108791030708445224/1108791030708445224

**Comment by pradeepb28** [comment]
> Hello guys, I was having the same issue and this really helps me to find out where the issue is. I hope it helps you too. https://discord.com/channels/839993398554656828/1108791030708445224/1108791030708445224

@adurov9 I am seeing this, would you mind sharing a screenshot
<img width="1261" alt="CleanShot 2023-12-13 at 11 03 09@2x" src="https://github.com/supabase-community/supabase-swift/assets/6823101/cae1a3d3-14dc-442a-b88f-249ebe09ec17">

**Comment by pradeepb28** [comment]
@grdsdev I see you have fixed it in #119 but is it available on Production now?

**Comment by grdsdev** [comment]
Hey @pradeepb28 yes, it is available, please use the latest 1.1.0 version.
